lib/Net/SIP/Packet.pm

```diff
@@ -648,7 +648,9 @@ sub as_parts {
 	my @header = split( m{\r?\n}, $header );
 
 	my $key2check;
-	if ( $header[0] =~m{^SIP/2.0\s+(\d+)\s+(\S.*?)\s*$} ) {
+	if ( !$header[0] ) {
+	    die "bad request: empty packet\n";
```

Comment:
Beware that this also matches non-empty packets containing string digits zero. Eg. `"000000000000000000"`.

It is better to check `if ( ! length $header[0] ) {`.

Reply:
Thanks for the suggestion, updated.